pkg/events/dependencies/manager.go

```diff
@@ -136,13 +121,22 @@ func (m *Manager) UnselectEvent(id events.ID) bool {
 // no longer valid).
 // It returns if managed to remove the event, as it might not be present in the tree.
 func (m *Manager) RemoveEvent(id events.ID) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	return m.removeEvent(id)
+}
+
+func (m *Manager) removeEvent(id events.ID) error {
```

Comment:
This is the dead lock fix. `RemoveEvent()`, a public method, was being called again after a lock by `removeEventDependents()`.